src/app/home-redesign/context.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import { ReactNode, createContext, useContext } from 'react';
+
+import { RecentBlocks } from './data';
+
+interface HomePageDataContextType {
+  recentBlocks: RecentBlocks;
+}
+
+const DEFAULT_HOME_PAGE_DATA: HomePageDataContextType = {
+  recentBlocks: {
+    btcBlocks: {
```

Comment:
I would love to see all of these names also include the prefix 'initial', so 'recentBlocks' becomes 'initialRecentBlocks'. This will make the logic inthe rest of the components easier to follow as we will clearly understand that this is just the initial data being used from the server to make the first render